changelog.d/16099.bugfix

```diff
@@ -0,0 +1 @@
+Prepare unit tests for Python 3.12.
```

Comment:
Since these are all in tests, I think `.misc` instead of `.bugfix` makes more sense (as server administrators won't really care about it).